src/main/java/org/icgc_argo/workflow/relay/service/IndexService.java

```diff
@@ -43,28 +45,38 @@ public IndexService(
   @SneakyThrows
   @StreamListener(IndexStream.WORKFLOW)
   public void indexWorkflow(JsonNode event) {
-    // TODO: Generalize event information parsing
+    // deserialize json events to metadata objects
+    val workflowEvent = MAPPER.treeToValue(event, WorkflowEvent.class);
```

Comment:
this can throw a de-serialization error that retrying won't solve, not sure what works best in your case for this kind of errors, but just flagging it